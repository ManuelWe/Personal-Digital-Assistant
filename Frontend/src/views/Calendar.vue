<template>
  <div
    id="help"
    class="hero-body"
  >
    <div class="container">
      <iframe
        id="idIframe"
        src="https://calendar.google.com/calendar/embed?src=rroff00labrg6qt5gu3ol87ejo%40group.calendar.google.com&ctz=Europe%2FBerlin"
        style="border: 0"
        width="1200"
        frameborder="0"
        scrolling="no"
        @onload="iframeLoaded()"
      />
    </div>
  </div>
</template>

<script>
export default {
  methods: {
    iframeLoaded() {
      const iFrameID = document.getElementById('idIframe');
      if (iFrameID) {
        // here you can make the height, I delete it first, then I make it again
        iFrameID.height = '';
        iFrameID.height = `${iFrameID.contentWindow.document.body.scrollHeight}px`;
      }
    },
  },
};
</script>

<style scoped>
iframe {
  height: 30rem;
}
.hero-body {
  position: relative;
  margin: 0rem 3rem 3rem 3rem;
  padding: 1rem 0 0 0;
}
</style>
