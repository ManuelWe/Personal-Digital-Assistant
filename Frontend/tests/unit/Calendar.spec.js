import { createLocalVue, mount } from '@vue/test-utils';
import Buefy from 'buefy';
import Calendar from '@/views/Calendar.vue';

const localVue = createLocalVue();
localVue.use(Buefy);

function factory() {
  return mount(Calendar, {
    localVue,
  });
}

describe('Calendar.vue', () => {
  describe('Elements are rendered correctly', () => {
    it('Component renders', () => {
      const wrapper = factory();
      expect(wrapper).toMatchSnapshot();
    });

    it('Loading Animation renders', () => {
      const wrapper = factory();
      expect(wrapper.find('.loading-overlay').exists()).toBe(true);
    });

    it('Calendar renders ', () => {
      const wrapper = factory();
      wrapper.vm.iframeLoading = false;
      wrapper.vm.$nextTick(() => {
        expect(wrapper.find('iframe').isVisible()).toBe(true);
      });
    });
  });

  it('Stop loading, if Calendar is loaded', () => {
    const wrapper = factory();
    wrapper.vm.$nextTick(() => {
      wrapper.find('iframe').trigger('load');
    });
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.iframeLoading).toBe(false);
    });
  });
});
