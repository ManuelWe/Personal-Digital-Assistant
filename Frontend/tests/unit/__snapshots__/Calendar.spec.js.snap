// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar.vue Elements are rendered correctly Component renders 1`] = `
<div id="help" class="hero-body">
  <div class="loading-overlay is-active">
    <div class="loading-background"></div>
    <div class="loading-icon"></div>
  </div>
  <div class="container"><iframe src="https://calendar.google.com/calendar/embed?src=rroff00labrg6qt5gu3ol87ejo%40group.calendar.google.com&amp;ctz=Europe%2FBerlin" width="1200" frameborder="0" scrolling="no" style="border: 0px; display: none;"></iframe></div>
</div>
`;
